code/parachain/frame/staking-rewards/src/lib.rs

```diff
@@ -715,75 +721,119 @@ pub mod pallet {
 			duration_preset: DurationSeconds,
 			keep_alive: bool,
 		) -> Result<Self::PositionId, DispatchError> {
-			let mut rewards_pool =
-				RewardPools::<T>::try_get(pool_id).map_err(|_| Error::<T>::RewardsPoolNotFound)?;
-
-			let reward_multiplier = Self::reward_multiplier(&rewards_pool, duration_preset)
-				.ok_or(Error::<T>::NoDurationPresetsConfigured)?;
+			RewardPools::<T>::try_mutate(pool_id, |maybe_rewards_pool| {
```

Comment:
What is the advantage of this restructuring? This actually reduces readability for me.

Reply:
`try_mutate` is the standard way to update something in storage, rather than reading and overwriting.

Reply:
Looking at the  storage impl it's basically the same steps for both. So I actually prefer the previous version here as it is less nested and easier to read.

Reply:
read/overwrite also introduces room for bugs (overwriting the wrong key, forgetting to insert at the end, inserting at the wrong time, etc etc), and while these would ideally be caught by tests, I prefer to not rely on tests to verify correctness. 

I also personally prefer the nested `try_mutate`s and I actually find it to be more readable as it describes the functionality better (mutate being one operation vs read/overwrite being two). When I see `Storage::{try_}mutate`, I know what the behaviour is - find the value in storage and mutate it somehow; whereas when I see a `get` or `try_get`, I don't know if it's just a read or if it's intended to be overwritten until I've read the entire function body.

Reply:
Also, I would like to eventually move the actual functionality/ trait impls out of the pallet itself, which would help reduce the indentation level

Reply:
> read/overwrite also introduces room for bugs (overwriting the wrong key, forgetting to insert at the end, inserting at the wrong time, etc etc) ..

I'm not convinced by this argument tbh. Those are standard things in normal code bases verified by business logic tests. Let's keep things flat as much as possible. If there is a way to use "try_mutate" without entire function body being inside the closure that's also fine though.